tuf/tuf_test.go

```diff
@@ -953,3 +953,143 @@ func TestGetAllRoles(t *testing.T) {
+	assert.NoError(t, err)
+	assert.Empty(t, delgPathPrefixes)
+	assert.Equal(t, []string{"path", "anotherpath"}, delgPaths)
+	assert.Equal(t, testKey1, delgRole.Keys[testKey1.ID()])
+
+	testKey2, err := ed25519.Create("targets/a", data.ED25519Key)
+	assert.NoError(t, err)
+	role, err = data.NewRole(
+		"targets/a", 1, []string{testKey2.ID()}, []string{""}, []string{})
+	assert.NoError(t, err)
+
+	err = repo.UpdateDelegations(role, data.KeyList{testKey2})
+	assert.NoError(t, err)
+
+	delgRole, err = repo.GetDelegationRole("targets/a")
+	assert.NoError(t, err)
+	assert.True(t, delgRole.IsDelegationRole())
+	assert.False(t, delgRole.IsBaseRole())
+	assert.Equal(t, "targets/a", delgRole.GetName())
+	assert.Equal(t, 1, delgRole.GetThreshold())
+	assert.Equal(t, []string{testKey2.ID()}, delgRole.ListKeyIDs())
+	delgPaths, err = delgRole.ListPaths()
+	assert.NoError(t, err)
+	assert.Equal(t, []string{""}, delgPaths)
+	delgPathPrefixes, err = delgRole.ListPathHashPrefixes()
+	assert.NoError(t, err)
+	assert.Empty(t, delgPathPrefixes)
+	assert.Equal(t, testKey2, delgRole.Keys[testKey2.ID()])
+
+	testKey3, err := ed25519.Create("targets/test/b", data.ED25519Key)
+	assert.NoError(t, err)
+	role, err = data.NewRole(
+		"targets/test/b", 1, []string{testKey3.ID()}, []string{"path/subpath", "anotherpath"}, []string{})
+	assert.NoError(t, err)
+
+	err = repo.UpdateDelegations(role, data.KeyList{testKey3})
+	assert.NoError(t, err)
+
+	delgRole, err = repo.GetDelegationRole("targets/test/b")
+	assert.NoError(t, err)
+	assert.True(t, delgRole.IsDelegationRole())
+	assert.False(t, delgRole.IsBaseRole())
+	assert.Equal(t, "targets/test/b", delgRole.GetName())
+	assert.Equal(t, 1, delgRole.GetThreshold())
+	assert.Equal(t, []string{testKey3.ID()}, delgRole.ListKeyIDs())
+	delgPaths, err = delgRole.ListPaths()
+	assert.NoError(t, err)
+	assert.Equal(t, []string{"path/subpath", "anotherpath"}, delgPaths)
+	delgPathPrefixes, err = delgRole.ListPathHashPrefixes()
+	assert.NoError(t, err)
+	assert.Empty(t, delgPathPrefixes)
+	assert.Equal(t, testKey3, delgRole.Keys[testKey3.ID()])
+}
+
+func TestGetDelegationRolesInvalidPaths(t *testing.T) {
+	ed25519 := signed.NewEd25519()
+	keyDB := keys.NewDB()
+	repo := initRepo(t, ed25519, keyDB)
+
+	testKey1, err := ed25519.Create("targets/test", data.ED25519Key)
+	assert.NoError(t, err)
+	role, err := data.NewRole(
+		"targets/test", 1, []string{testKey1.ID()}, []string{"path", "anotherpath"}, []string{})
+	assert.NoError(t, err)
+
+	err = repo.UpdateDelegations(role, data.KeyList{testKey1})
+	assert.NoError(t, err)
+
+	testKey2, err := ed25519.Create("targets/test/b", data.ED25519Key)
+	assert.NoError(t, err)
+	// Now we add a delegation with a path that is not prefixed by its parent delegation
+	role, err = data.NewRole(
+		"targets/test/b", 1, []string{testKey2.ID()}, []string{"invalidpath"}, []string{})
+	assert.NoError(t, err)
+
+	err = repo.UpdateDelegations(role, data.KeyList{testKey2})
```

Comment:
ah this is related to what I just commented -- yeah I think it'd be better to have this in a separate PR.

It's actually an interesting case because this method doesn't use a `Set...` from the tuf repo.  This function could probably benefit from being refactored to use a `CheckAndSet...` function